components/AdminPanel/AdminPanel.tsx

```diff
@@ -1,3 +1,83 @@
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useCallback } from 'react';
+
+import { useQuery } from '../../hooks/useQuery';
+import { getBlogs } from '../../utils/api/getBlogs';
+import { formatDate } from '../../utils/date-utils';
+import { addTrackingToLink } from '../../utils/link-utils';
+import { BlogsTable } from '../BlogsTable/BlogsTable';
+
+import styles from './adminPanel.module.scss';
+
 export const AdminPanel = () => {
-  return <section>admin panel</section>;
+  const router = useRouter();
+  const publicQuery = window?.location.search || '';
+
+  const { value: blogs } = useQuery(useCallback(() => getBlogs(publicQuery), [publicQuery]));
+
+  if (!blogs) {
+    return <p>Ładowanie...</p>;
+  }
+
+  const tableData = blogs.map((blog) => ({
```

Comment:
możesz użyć destrukturyzacji tutaj 
```suggestion
  const tableData = blogs.map(({createdAt, lastArticlePublishedAt }) => ({
```

Reply:
Myślę, że mija się to z celem, ze zbyt wielu właściwości później korzystam.